components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/statshandler/GatewayStatsUpdater.java

```diff
@@ -0,0 +1,108 @@
+        log.debug("Updated stats publishing status in Gateway to : " + statUpdateStatus);
+
+        ServiceDataPublisherAdmin serviceDataPublisherAdmin = APIManagerComponent.getDataPublisherAdminService();
+        EventingConfigData eventingConfigData = null;
+        if (serviceDataPublisherAdmin != null) {
+            eventingConfigData = serviceDataPublisherAdmin.getEventingConfigData();
+        }
+
+        try {
+            if (eventingConfigData != null) {
+                //config values are updated if the stats publishing is true
+                if (statUpdateStatus) {
+                    log.debug("Updating values related to stats publishing status.");
+                    //values related to stats publishing status are only updated if all of them are non-empty
+                    if (!(receiverUrl.isEmpty()) && !(user.isEmpty()) && !(password.isEmpty())) {
+                        analyticsConfiguration.setBamServerUrlGroups(receiverUrl);
+                        analyticsConfiguration.setBamServerUser(user);
+                        analyticsConfiguration.setBamServerPassword(password);
+
+                        eventingConfigData.setUrl(receiverUrl);
+                        eventingConfigData.setUserName(user);
+                        eventingConfigData.setPassword(password);
+                        log.debug("BAMServerURL : " + receiverUrl + " , BAMServerUserName : " + user + " , " +
+                                "BAMServerPassword : " + password);
+                        APIUtil.addBamServerProfile
+                                (receiverUrl, user, password, MultitenantConstants.SUPER_TENANT_ID);
+                    }
+                }
+                //eventingConfigData must be updated irrespective of the value of statUpdateStatus
+                eventingConfigData.setServiceStatsEnable(statUpdateStatus);
+
+                //this may throw an Exception
+                serviceDataPublisherAdmin.configureEventing(eventingConfigData);
+            }
+            //send the cluster message to other nodes in the cluster to update stats publishing status
+            ClusteringAgent clusteringAgent = this.getConfigContext().getAxisConfiguration().getClusteringAgent();
+            if (clusteringAgent != null) {
+                log.debug("Sending cluster message to Gateway domain to update stats publishing status.");
+                clusteringAgent.sendMessage(new StatClusterMessageUtil(statUpdateStatus), true);
+            }
+        } catch (ClusteringFault clusteringFault) {
+            //error is only logged because initial gateway has modified the status
+            log.error("Failed to send cluster message to Gateway domain and update stats publishing status.");
+        } catch (APIManagementException e) {
+            log.error("Error occurred while adding BAMServerProfile");
+        } catch (Exception e) {
+            log.error("Error occurred while updating EventingConfiguration");
```

Comment:
Also mention what can be the outcome of a failure. What functionality can get impaired.